### PR TITLE
add meta

### PR DESCRIPTION
**Pull request by bagari**
MeTA, which stands for [Medial Tractography Analysis](https://github.com/bagari/meta), is a workflow designed to minimize brain microstructural heterogeneity in diffusion MRI (dMRI) metrics by extracting and parcellating the core volume along the length of white matter bundles in voxel-space, directly preserving bundle shape while efficiently capturing regional variation within and along the bundles. MeTA is applied to the study of human brain diseases and the genetic architecture of white matter, providing valuable insights into both neurodegenerative conditions and the structural components of brain connectivity.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/meta/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+{% set version = "1.0.0" %}
+{% set sha256 = "96673754c8f8c926bd179a824513de36d0b361eb93b1f7781cbac27ecf520d6e" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/bagari/meta/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
```
[the remark this answers is not in the document]
Comment (bagari) [reply recipes/meta/meta.yaml:11]: Updated package name to meta

File: recipes/meta/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
```
[the remark this answers is not in the document]
Comment (bagari) [reply recipes/meta/meta.yaml]: Updated package name to meta

File: recipes/meta/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    - libexpat >=2.5.0,<3.0a0
+    - vtk-io-ffmpeg >=9.2.6,<9.3.0 build=*{{ build_variant }}*
+
+test:
+  imports:
+    - meta
+  commands:
+    - meta --help
+    - cmrep_vskel --version
+    - vtklevelset --version
```
[the remark this answers is not in the document]
Comment (bagari) [reply recipes/meta/meta.yaml]: I added small data to perform comprehensive import tests.

File: recipes/meta/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    - vtk-io-ffmpeg >=9.2.6,<9.3.0 build=*{{ build_variant }}*
+
+test:
+  source_files:
+    - resources/CST_R.nii.gz
+  commands:
+    - meta --help
+    - vtklevelset resources/CST_R.nii.gz CST_R.vtk 0.1
+    - cmrep_vskel -c 3 -p 1.5 -g CST_R.vtk CST_R_skeleton.vtk
+
```
[the remark this answers is not in the document]
Comment (bagari) [reply recipes/meta/meta.yaml:79]: updated

File: recipes/meta/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    - nibabel
+    - pandas
+    - pyvista
+    - scipy
+    - tqdm
+    - tslearn
+    - dipy
+    - libexpat <2.6
+    - libexpat >=2.5.0,<3.0a0
+    - vtk-io-ffmpeg >=9.2.6,<9.3.0 build=*{{ build_variant }}*
```
[the remark this answers is not in the document]
Comment (bagari) [reply recipes/meta/meta.yaml:70]: itk == 5.3.0 is import to build c++ code

File: recipes/meta/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    - pip
+    - boost = 1.82
+    - itk == 5.3.0
+    - libitk == 5.3.0
+    - libitk-devel == 5.3.0
+    - hdf5 >=1.14.2,<1.14.3.0a0
+    - vtk-base >=9.2.6,<9.3.0 build=*{{ build_variant }}*
+    - libexpat >=2.5.0,<2.6.0
+    - vtk-io-ffmpeg >=9.2.6,<9.3.0 build=*{{ build_variant }}*
+
```
[the remark this answers is not in the document]
Comment (bagari) [reply recipes/meta/meta.yaml:33]: I do need pip and Python as the package needs to compile C++ and then will install Python code.

File: recipes/meta/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+test:
+  source_files:
+    - resources/CST_R.nii.gz
+  imports:
+    - meta
+  commands:
+    - meta --help 2>&1 || exit 1
+    - vtklevelset resources/CST_R.nii.gz CST_R.vtk 0.1 && test -f CST_R.vtk
+    - cmrep_vskel -c 3 -p 1.5 -g CST_R.vtk CST_R_skeleton.vtk && test -f CST_R_skeleton.vtk
+
```
[the remark this answers is not in the document]
Comment (bagari) [reply recipes/meta/meta.yaml:77]: - meta.core and - meta.utils does not exist in my package

File: recipes/meta/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    - qhull
+    - boost = 1.82
+    - itk == 5.3.0
+    - libitk == 5.3.0
+    - libitk-devel == 5.3.0
+    - hdf5 >=1.14.2,<1.14.3.0a0
+    - vtk-base >=9.2.6,<9.3.0 build=*{{ build_variant }}*
+    - libexpat >=2.5.0,<2.6.0
+    - vtk-io-ffmpeg >=9.2.6,<9.3.0 build=*{{ build_variant }}*
+
```
[the remark this answers is not in the document]
Comment (bagari) [reply recipes/meta/meta.yaml:46]: It's okay for now, and the code should be compiled successfully

File: recipes/meta/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+    - python >=3.8,<3.12
+    - pip
+    - numpy >=1.21
+    - nibabel
+    - pandas
+    - pyvista
+    - scipy
+    - tqdm
+    - tslearn
+    - dipy
```
[the remark this answers is not in the document]
Comment (bagari) [reply recipes/meta/meta.yaml:68]: The build is being terminated after ~30 minutes despite bulk: true

**Comment by bagari** [comment]
@BiocondaBot please add label

**Review by martin-g — APPROVED** [review]

**Comment by martin-g** [comment]
It seems there was a PR check failure: https://github.com/bioconda/bioconda-recipes/actions/runs/11566537250/job/32195443196#step:7:44
I am not sure whether this caused the problem with the release...

**Comment by bagari** [comment]
@martin-g I'm not sure if this https://github.com/bioconda/bioconda-recipes/actions/runs/11566537250/job/32195443196#step:7:44
refer to my package. I could update the package name if it creates confusion with meta.yaml. I don't think that I have "`datasets >=2`"
```
Error: bad character '>=' in package version dependency 'datasets'
Perhaps you meant 'datasets >=2'
```

**Comment by martin-g** [comment]
See https://github.com/bioconda/bioconda-recipes/issues/33333#issuecomment-2442652599

**Pull request merged**